packages/app/src/pages/Profile.tsx

```diff
@@ -19,12 +30,58 @@ export const Profile = (): JSX.Element => {
       title: '',
     },
   });
-  const { provider } = usePractitioner();
+
+  useEffect(() => {
+    reset({
+      ...provider,
+    });
+  }, [provider, reset]);
+
   const onSubmit = (data: FormData): void => {
-    console.log(data);
-    // TODO: form submission structure
+    updatePractitioner(data).catch((error) => {
+      console.log(error);
+      setOpenSnackbar(true);
+    });
+  };
+
+  const handleCloseSnackbar = (): void => {
+    setOpenSnackbar(false);
   };
 
+  async function updatePractitioner(data: FormData): Promise<void> {
+    const accessToken = await getAccessTokenSilently();
+    providerPatchOps.push(
+      {
+        op: practitionerProfile?.identifier?.[0].value ? 'replace' : 'add',
+        path: '/identifier/0/value',
+        value: data.slug,
+      },
+      {
+        op: practitionerProfile?.name?.[0].given?.[0] ? 'replace' : 'add',
+        path: '/name/0/given/0',
+        value: data.firstName,
+      },
+      {
+        op: practitionerProfile?.name?.[0].family ? 'replace' : 'add',
+        path: '/name/0/family',
+        value: data.lastName,
+      },
+      {
+        op: practitionerProfile?.name?.[0].prefix?.[0] ? 'replace' : 'add',
+        path: '/name/0/prefix/0',
+        value: data.title,
+      }
+    );
+
+    updateProvider(accessToken, practitionerProfile?.id || '', providerPatchOps).catch((error) => {
+      console.log(error);
```

Comment:
It seems like if this request failed for some reason, we'd catch it here and then not show the user the error. Should consider rethrowing and handle on the outer function call.